src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -441,6 +441,10 @@ private function dehydrateValue(mixed $value, LivePropMetadata $propMetadata, ob
             if ($propMetadata->collectionValueType() && Type::BUILTIN_TYPE_OBJECT === $propMetadata->collectionValueType()->getBuiltinType()) {
                 $collectionClass = $propMetadata->collectionValueType()->getClassName();
                 foreach ($value as $key => $objectItem) {
+                    if (null !== $collectionClass && null === $objectItem) {
+                        $objectItem = new $collectionClass();
```

Comment:
This will break as soon as constructor will have arguments. (yes we have some DTO's with constructor args) :)

Reply:
Having DTO or Entity with constructor is already not allowed in this package:

Look at the follow code in DoctrineEntityHydrationExtension.php; line 38

```php
public function hydrate(mixed $value, string $className): ?object
    {
        // an empty array means a non-persisted entity
        // we support instantiating with no constructor args
        if (\is_array($value) && 0 === \count($value)) {
            return new $className();
        }
```

Also in de hydrateObjectValue function in LiveComponentHydrator.php; line: 498

```php
if (\is_array($value)) {
  $object = new $className();
  foreach ($value as $propertyName => $propertyValue) {
      $reflectionClass = new \ReflectionClass($className);
      $property = $reflectionClass->getProperty($propertyName);
      $propMetadata = $this->liveComponentMetadataFactory->createLivePropMetadata($className, $propertyName, $property, new LiveProp());
      $this->propertyAccessor->setValue($object, $propertyName, $this->hydrateValue($propertyValue, $propMetadata, $component));
  }
  
  return $object;
}
```

Reply:
So that is for the Entity construction, but not the DTO

Reply:
The second example, LiveComponentHydrator.php, is not an Entity example but rather the default hydration function for the DTO.

$object = new $className() is also without arguments. If we allow this in hydration, it should not be a problem in the dehydration function and its resolved this issue.

Reply:
I think we can have a different solution. If your array is truly an array of DTO objects OR null values, then during dehydration, when we see a null value, we can keep that value as null. Then during hydration, when we see the null value in the collection, once again, we keep it as null. This means you get back the exact same array after dehydration & hydration as you had originally.